spec/integration/user_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+
+RSpec.describe 'User integration tests', type: :feature do
+  before(:each) do
+    @user1 = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo1',
+                         bio: 'Teacher from Mexico.', posts_counter: 4)
+    @user2 = User.create(name: 'John', photo: 'https://unsplash.com/photos/F_-0BxGuVvo2',
+                         bio: 'Teacher from Mexico.', posts_counter: 7)
+    @user3 = User.create(name: 'Mary', photo: 'https://unsplash.com/photos/F_-0BxGuVvo3',
+                         bio: 'Teacher from Mexico.', posts_counter: 2)
+    @users = [@user1, @user2, @user3]
+  end
+  describe 'index' do
+    before { visit users_path }
+    it 'should display the username of all other users' do
+      @users.each do |user|
+        expect(page).to have_content(user.name)
+      end
+    end
+    it 'should display the profile picture for each user' do
+      @users.each do |user|
+        expect(page).to have_selector("img[src='#{user.photo}']")
+      end
+    end
+    it 'should display the number of post for each user' do
+      @users.each do |user|
+        expect(page).to have_content("Number of posts: #{user.posts_counter}")
+      end
+    end
+    it 'should redirect to the show user page when clicking on the user' do
+      @users.each do |user|
+        click_link(user.name)
+        expect(page).to have_content(user.name)
+        visit users_path
+      end
+    end
+  end
+  describe 'show' do
+    before(:each) do
```

Comment:
- [ ] Dear Team, Please notice we need a separate file to test the `User index` and `User show` pages, Please create a separate file for the `User index` and `User show` tests.